call_host.sh

```diff
@@ -8,15 +9,70 @@ if [ -f "$CALL_HOST_CONFIG" ]; then
 	source "$CALL_HOST_CONFIG"
 fi
 
+# zsh / bash compatibility helpers
+is_zsh(){
+	# detect the current shell process name (portable ps usage)
+	if command -v ps >/dev/null 2>&1; then
+		# get last path component if ps returns full path
+		p="$(ps -p $$ -o comm= 2>/dev/null | awk -F/ '{print $NF}')"
+		case "$p" in
+			zsh) return 0 ;;
+		esac
+	fi
+
+	# fallback: check common names for $0 or $ZSH_NAME (login shells may have a leading dash)
+	case "$(basename -- "${ZSH_NAME:-$0}" 2>/dev/null)" in
+		zsh|-zsh) return 0 ;;
+	esac
+
+	return 1
+}
+
+if is_zsh; then
+	# export a function to the environment for child shells (zsh)
+	export_func(){
+		typeset -fx "$1" 2>/dev/null || true
+	}
+	# declare an associative array (zsh) - create a named array using eval so dynamic name works
+	declare_assoc(){
+		eval "typeset -A $1"
+	}
+	# get current function name in zsh (be tolerant if indices differ)
+	current_funcname(){
+		echo "${funcstack[1]:-${funcstack[0]:-}}"
+	}
+else
+	# bash
+	export_func(){
+		[ -n "$1" ] || return
+		eval "export -f $1" 2>/dev/null || true
```

Comment:
`eval` is not necessary here. The shellcheck warning should be handled in either of the following ways:
1. put `# shellcheck disable=SC2163` above this line
2. use `export -f ${1?}` as noted at https://www.shellcheck.net/wiki/SC2163